lib/services/database.go

```diff
@@ -641,6 +839,34 @@ func IsMemoryDBClusterSupported(cluster *memorydb.Cluster) bool {
 	return aws.BoolValue(cluster.TLSEnabled)
 }
 
+func IsAzureMySQLServerAvailable(state armmysql.ServerState) bool {
+	switch state {
+	case armmysql.ServerStateReady:
+		return true
+	case armmysql.ServerStateInaccessible,
+		armmysql.ServerStateDropping,
+		armmysql.ServerStateDisabled:
+		return false
+	default:
+		log.Warnf("Unknown Azure MySQL server state: %q", state)
+		return false
+	}
+}
+
+func IsAzurePostgresServerAvailable(state armpostgresql.ServerState) bool {
+	switch state {
+	case armpostgresql.ServerStateReady:
+		return true
+	case armpostgresql.ServerStateInaccessible,
+		armpostgresql.ServerStateDropping,
+		armpostgresql.ServerStateDisabled:
+		return false
+	default:
+		log.Warnf("Unknown Azure Postgres server state: %q", state)
+		return false
+	}
+}
```

Comment:
If the statuses are the same maybe we can consolidate the `IsAzureMySQLServerAvailable` and IsAzurePostgresServerAvailable into one function. Also remember to add godocs comment

Reply:
Yes, done in refactor